send_message_msteams_and_aws_script.py

```diff
@@ -0,0 +1,277 @@
+        myTeamsMessage = pymsteams.connectorcard(self.webhook_url)
+        myTeamsMessage.color(self.MESSAGE_COLOR)
+        myTeamsMessage.title(title)
+        myTeamsMessage.text(
+            f"**Quote:** {quote_text}\n\n"
+            f"**Author:** {quote_author}\n\n"
+            f"![Image]({image_url})"
+        )
+
+        try:
+            myTeamsMessage.send()
+            logging.info("Message successfully sent.")
+            return
+        except (ConnectionError, HTTPError, TimeoutError) as e:
+            logging.error(
+                f"Failed to send message due to network error: {e}",
+                exc_info=True
+            )
+        except pymsteams.TeamsWebhookException as e:
+            logging.error(
+                f"Failed to send message due to Teams webhook error: {e}",
+                exc_info=True
+            )
+        except ValueError as e:
+            logging.error(
+                f"Failed to send message due to value error: {e}",
+                exc_info=True
+            )
+
+
+class S3Loader:
+    def __init__(self):
+        self.s3_key = None
+        self.s3_client = boto3.client(
+            service_name='s3',
+            aws_access_key_id='test',
+            aws_secret_access_key='test',
+            endpoint_url='http://localhost:4566',
+        )
+
+        self.bucket_name = "images-msteams"
+
+    def extract_image_name(self, url):
+        parsed_url = urlparse(url)
+        img_name = parsed_url.path.split("/")[-1]
+        return img_name
+
+    def image_upload(self, image_url):
+        try:
+            response = requests.get(image_url)
+            response.raise_for_status()
+            image_data = response.content
+
+            with BytesIO() as f:
+                f.write(image_data)
+
+            timestamp = datetime.now().strftime("%Y%m%d%H%M%S")
+
+            image_name_from_url = self.extract_image_name(image_url)
+
+            image_name = f"{image_name_from_url}_{timestamp}.jpg"
+            s3_key = f"{image_name}"
+
+            self.s3_client.put_object(
+                Bucket=self.bucket_name,
+                Key=s3_key,
+                Body=image_data
+            )
+            s3_image_url = f"{self.s3_client.meta.endpoint_url}/{self.bucket_name}/{s3_key}"
+
+            logging.info(f"Image successfully uploaded to S3: {s3_image_url}")
+            return s3_image_url
+        except (BotoCoreError, NoCredentialsError) as e:
+            logging.error(
+                f"Failed to upload image to S3: {e}",
+                exc_info=True
+            )
+            return None
+
+
+class RDSLoader:
+    def __init__(self, host, port, dbname, user, password):
+        self.conn = psycopg2.connect(
+            host=host,
+            port=port,
+            dbname=dbname,
+            user=user,
+            password=password
+        )
+        self.check_table_exists()
+
+    def check_table_exists(self):
```

Comment:
I think that's a bad practice to execute table creation from a code - it is more controllable than it's done manually or with some migration tools (like yoyo)

Reply:
done